### PR TITLE
Stop requiring event args to extend EventArgs (for .NET Standard)

### LINKED ISSUE #361: stop requiring event arguments to extend EventArgs in event raisers for .NET Standard
From @bondsbw in #30:

> I know I'm coming in late to this discussion, but something to consider (I apologize if I missed this in past discussion):\
> .NET 4.5 no longer constrains EventHandler to using something that derives from EventArgs. TEventArgs can now be anything.
> 
> Of course compiling for .NET 4.5 may not make sense depending on the goals of the project.
> But I made a few alterations to the source to bump it to 4.5 and remove the constraint,
> and was able to get everything working as expected (other than Silverlight projects which I did not load).

### PR DESCRIPTION
**Pull request by blairconrad**
Fixes #361.

### CODE REVIEW

**Comment by blairconrad** [comment]
The first commit just pays off some tech debt - we weren't running all the specs we could.

File: tests/FakeItEasy.Specs/EventRaisingSpecs.cs [hunk 1]
```diff
@@ -146,6 +158,21 @@ public void EventArguments()
...
+            "when raising event passing arguments"
+                .x(() => Fake.EventHandlerOfNonEventArgsEvent += Raise.With(this.referenceTypeEventArgs));
+
+            "it should pass the fake as sender"
+                .x(() => CapturedSender.Should().BeSameAs(Fake));
+
+            "it should pass the event arguments"
+                .x(() => CapturedArgs1.Should().BeSameAs(this.referenceTypeEventArgs));
+        }
+#endif
```
Comment (thomaslevesque) [inline tests/FakeItEasy.Specs/EventRaisingSpecs.cs]: These specs should be updated to GWT style

File: tests/FakeItEasy.Specs/FakeItEasy.Specs.csproj [hunk 2]
```diff
@@ -24,7 +24,7 @@
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>bin\Debug\</OutputPath>
-    <DefineConstants>TRACE;DEBUG;FEATURE_SELF_INITIALIZED_FAKES</DefineConstants>
+    <DefineConstants>TRACE;DEBUG;FEATURE_SELF_INITIALIZED_FAKES FEATURE_EVENT_ARGS_MUST_EXTEND_EVENTARGS</DefineConstants>
```
Comment (thomaslevesque) [inline tests/FakeItEasy.Specs/FakeItEasy.Specs.csproj]: Nitpick: there's a inconsistency here... some symbols are separated by `';'`, others by `' '`

File: tests/FakeItEasy.Specs/EventRaisingSpecs.cs (hunk 1 above)
Comment (blairconrad) [inline tests/FakeItEasy.Specs/EventRaisingSpecs.cs]: Of course. Sorry/thanks!

File: tests/FakeItEasy.Specs/FakeItEasy.Specs.csproj (hunk 2 above)
Comment (blairconrad) [inline tests/FakeItEasy.Specs/FakeItEasy.Specs.csproj]: I noticed, but when I looked at the help, it told me to separate by spaces, but the IDE added the semicolons after TRACE and DEBUG. I now see I was looking at the VS2012 help (why this is the default page when I hit F1 in VS2015, I could not tell you). 2015's help does say to use semicolons. Will fix.

Comment (thomaslevesque) [inline tests/FakeItEasy.Specs/FakeItEasy.Specs.csproj]: Not a big deal, since it works with either space or semicolons...

**Comment by blairconrad** [comment]
* fixup:
    * semicolons, not spaces
    * in the same fixup, moved the new spec - I think it's better lower down
* GWT syntax
* a little more tidying in the specs - whitespace, removing a weird exception-catching method, that sort of thing

File: tests/FakeItEasy.Specs/EventRaisingSpecs.cs [hunk 3]
```diff
@@ -189,136 +204,167 @@ public void NullSenderAndEventArguments()
...
+            "And the supplied value is passed as the event arguments"
                 .x(() => CapturedArgs1.Should().BeSameAs(this.customEventArgs));
         }
 
         [Scenario]
         public void CustomEventHandler()
         {
-            "when raising custom event passing sender and arguments"
+            "Given an event of type EventHandler with custom event arguments type"
+                .See(() => nameof(Fake.GenericEvent));
```
Comment (thomaslevesque) [inline tests/FakeItEasy.Specs/EventRaisingSpecs.cs]: It should be `CustomEvent` here, not `GenericEvent` (the description also needs to be changed)

Comment (blairconrad) [inline tests/FakeItEasy.Specs/EventRaisingSpecs.cs]: Whoops. So it should. Thanks! I think the test should appear below the new one as well.

**Comment by blairconrad** [comment]
Thanks, @thomaslevesque. Rebase was more work than anticipated. I guess reordering members and touching half the lines in a file don't mix well. :)

While rebasing, I noticed we still had semicolons and spaces in FakeItEasy.csproj, so I fixed while I was at it. Too impatient to wait another round of review, so I squashed that right in. Here's the diff between what you reviewed and what I'm pushing:

```diff

λ  git diff origin/no-eventargs-netstd
diff --git a/src/FakeItEasy/FakeItEasy.csproj b/src/FakeItEasy/FakeItEasy.csproj
index 3e8ecb2..c241037 100644
--- a/src/FakeItEasy/FakeItEasy.csproj
+++ b/src/FakeItEasy/FakeItEasy.csproj
@@ -30,7 +30,7 @@
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>bin\Debug\</OutputPath>
-    <DefineConstants>TRACE;DEBUG;FEATURE_BINARY_SERIALIZATION FEATURE_REFLECTION_GETASSEMBLIES FEATURE_SELF_INITIALIZED_FAKES FEATURE_EVENT_ARGS_MUST_EXTEND_EVENTARGS</DefineConstants>
+    <DefineConstants>TRACE;DEBUG;FEATURE_BINARY_SERIALIZATION;FEATURE_REFLECTION_GETASSEMBLIES;FEATURE_SELF_INITIALIZED_FAKES;FEATURE_EVENT_ARGS_MUST_EXTEND_EVENTARGS</DefineConstants>^M
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
     <CodeAnalysisRuleSet>..\..\FakeItEasy.ruleset</CodeAnalysisRuleSet>
@@ -41,7 +41,7 @@
     <DebugType>pdbonly</DebugType>
     <Optimize>true</Optimize>
     <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE;FEATURE_BINARY_SERIALIZATION FEATURE_REFLECTION_GETASSEMBLIES FEATURE_SELF_INITIALIZED_FAKES FEATURE_EVENT_ARGS_MUST_EXTEND_EVENTARGS</DefineConstants>
+    <DefineConstants>TRACE;FEATURE_BINARY_SERIALIZATION;FEATURE_REFLECTION_GETASSEMBLIES;FEATURE_SELF_INITIALIZED_FAKES;FEATURE_EVENT_ARGS_MUST_EXTEND_EVENTARGS</DefineConstants>^M
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
     <DocumentationFile>Bin\Release\FakeItEasy.xml</DocumentationFile>
```

**Comment by thomaslevesque** [comment]
Thanks @blairconrad!

**Comment by blairconrad** [comment]
Ha, thank you.

**Pull request merged**